### ISSUE TITLE
Merging doesn't finish

### ISSUE THREAD

**Issue by paddotk** [dawnlabs/alchemy#53]
Windows 10, using Alchemy version 0.5.0

When I drag and drop some files (.png) into the application, fill in a title, and click the start button, the 'in progress' screen is shown, and it stays on there. No pdf is produced and there is no way to cancel it.

Additionally, it seems it's not possible to exit the application at any point. I'm forced to end all Electron processes in task manager.

**Comment by mfix22** [comment]
@ericadamski do you have access to a Windows computer to test this case out on?

**Comment by ericadamski** [comment]
@mfix22 I am sorry but I do not 😞 This is just a windows issue?

**Comment by mfix22** [comment]
No, we don't for 0.5.1, but you could build and run the project pretty easily if you clone this repo, and then run `yarn && yarn build && yarn start`

**Comment by paddotk** [comment]
Aight, I'll give it a spin

**Comment by paddotk** [comment]
@mfix22 I tried it (kinda forgot so it took a bit long, sorry 😛) and unfortunately, the latest build loses focus and minimizes to the task bar, as soon as a click anywhere outside the app window. Hence I can't actually drag any files onto it.

The option to close the app is also missing here btw.

**Comment by ericadamski** [comment]
Could it be a difference on how menu bar applications work on Windows machines?

**Comment by mfix22** [comment]
> Could it be a difference on how menu bar applications work on Windows machines? 
Could be! I am not sure 😬

**Comment by paddotk** [comment]
It does work a bit differently than OSX or Linux, yeah. There are the normal buttons that indicated any open applications. But also, at the right there are small icons for notifications and apps that run in the background, and that's where the Alchemy icon is shown. You can't open this by dragging a file over that, I'm afraid.

![image](https://user-images.githubusercontent.com/11683974/48936663-e88c7d00-ef0b-11e8-9f37-16c75539d729.png)

**Comment by chucklepie** [comment]
Hello,
I'm on Windows 10, downloaded the latest verison. Sorry, no idea what version because the download site contains no information and neither does this github site (that I can see easily).

Drag a simple file, click merge, and all I get is 'Converting'. There is no config I can see to add logging, there is no log file, just nothing.

This is 2 years since the original post, so I guess this software is not updated or supported anymore?

**Comment by ericadamski** [comment]
@chucklepie Thanks for brining this up!

I unfortunately do not have access to a Windows machine to test this out. Can you attach a video recording of the error happening? Maybe I can figure out some issues from there.

**Comment by chucklepie** [comment]
Nothing happens, literally, it just says converting and does nothing. No
need for a video.

If there was a logger added to the code it might help...

**Comment by PentagramPro** [comment]
Same for me, Windows 10:

1. Start the app, click the icon in tray to open window
2. Drag and drop files, click 'Merge'
Result:
It starts converting and never finishes

**Comment by DLandDS** [comment]
Same for me too, Windows 10:
it's converting forever

**Comment by binlabs** [comment]
It sounds like @DLandDS and @PentagramPro and @chucklepie are seeing more than I am in terms of the UI. Here is what I experienced in Windows 10:

1. The window loses focus and minimizes the second you try to drag any kind of file, and does not re-appear when you drag the file over the icon in the taskbar. Many people have reported this. However, a workaround I have found (and this is actually kind of difficult) is to start dragging the file, use the **CTRL+SHIFT+8** shortcut built-in to Alchemy to toggle the window, and finally drop the file.
2. I don't get any kind of menu, options, or messages anywhere; not by clicking the icon, right-clicking, not in the console - nowhere.
3. HOWEVER, it does successfully convert the file for me. I don't get any kind of options what it converts to, but I successfully dropped a PNG from my desktop and ended up with a converted PDF on my desktop. I assume PDF must be the default option or something, I wouldn't know because I see literally nothing else but the icon in the taskbar and the Dropzone window when I toggle it by left-clicking the icon or by using the keyboard shortcut I mentioned above.

To get the build working at all, I was forced to update the execa dependency from `execa@^0.6.3` in package.json to `execa@^4.0.0`. I was getting some type of filesystem-related async error that I can't quite recall at the moment, and searching for that error suggested execa was the culprit, and it was. I also can't recall if this caused the built to fail, or if it caused the start to fail, sorry for the keeping track of that, but I could reproduce if needed.

As soon as I get some free time I'll take a video of what I do and what I don't see in hopes that someone might have some suggestions.

It seems pretty clear to me that in order to get Alchemy functioning in Windows as well as it does in OSX, there are going to need to be some pretty big changes that apply only to the win32 `process.platform`, and that a lot of the changes are likely to be visual, as Windows just doesn't support the ability to drag a file on top of an icon to feed it to that application, and because dragging a file on top of an icon does not toggle the visibility of the window.

I don't currently know enough about Electron to make these specific changes myself right now, but I'll be looking into it as time allows.

**Comment by Jaja0624** [comment]
@binlabs You can fix your issue #1 by commenting out line this line in main.js 

`mb.on('focus-lost', () => mb.hideWindow());`

unfortunately I am currently just getting a blank white screen when I open the app. I updated execa as well. I'm trying to play with the source code to see if I can fix the app (0.50 release) not being able to convert on windows.

**Comment by mfix22** [comment]
Hey folks, I appreciate all your discussion around this issue, but I have to say that I no longer have time to support this project (thus https://github.com/dawnlabs/alchemy/issues/32). 

Would be happy to help merge in PRs, etc, and even support a new version with all our dependencies upgraded and polished, but I won't have any time to actively work on this project 😞
